### DISCUSSION TITLE
how to get issue types by Github App？

### DISCUSSION THREAD

**Discussion by herman-ola** [community/community#142232] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

how to get issue types by Github App，use GraphQL Api；
i am use Github app token request  GraphQL Api，response error 

request params:
{
"query":"query { organization(login: \"olaola-chat\"){issueTypes(first: 10) {pageInfo{endCursor,hasNextPage},nodes {id,name,isEnabled,isPrivate}}}}"
}

headers:
Authorization:bearer ghs_xxx (github app token)
GraphQL-Features:issue_types

response:
{
    "errors": [
        {
            "path": [
                "query",
                "organization",
                "issueTypes"
            ],
            "extensions": {
                "code": "undefinedField",
                "typeName": "Organization",
                "fieldName": "issueTypes"
            },
            "locations": [
                {
                    "line": 1,
                    "column": 44
                }
            ],
            "message": "Field 'issueTypes' doesn't exist on type 'Organization'"
        }
    ]
}

**Comment by ulyssecoispellier** [comment]
The error message you’re receiving indicates that the issueTypes field does not exist on the Organization type in GitHub’s GraphQL API. This suggests that either the field is not available or there is an issue with the query structure.

Here’s a breakdown of the issues and steps to fix it:

	1.	Field availability: The error suggests that issueTypes is not available under Organization. Currently, GitHub’s GraphQL API does not support fetching issue types directly at the organization level.
	2.	Where to fetch issue types: To fetch issue types (or issues themselves), you should typically query repositories rather than the organization directly. The issue types and issues are scoped at the repository level, not the organization level.
	3.	Query structure:
	•	If you want to query issues from repositories, you will need to go through the repository field and not the organization field.
	4.	Authorization: Ensure that your GitHub App has sufficient permissions to query repository issues.

Example Query for Issues in a Repository

To query issues or issue types (assuming you’re looking for issues themselves or custom issue types, a newer feature of GitHub), you would have to structure the query differently, like this:

```graphql
query {
  repository(owner: "olaola-chat", name: "your-repo-name") {
    issues(first: 10) {
      pageInfo {
        endCursor
        hasNextPage
      }
      nodes {
        id
        title
        number
        labels(first: 5) {
          nodes {
            name
          }
        }
        createdAt
        state
      }
    }
  }
}
```

This query retrieves the first 10 issues from a repository, along with their labels, creation date, and state.

Steps to resolve:

	1.	Check Permissions: Ensure that the GitHub App has permission to access repositories and issues. This is configured in the app’s settings under “Repository permissions” → “Issues”.
	2.	Modify the Query: If you’re looking for issues or custom issue types in a specific repository, modify the query to target the repository field instead of the organization.
	3.	Test the Query: Use GraphiQL or GitHub’s GraphQL explorer to test the query manually with your app token.
